### PR TITLE
xnresize: Add version 1.00

### PR DESCRIPTION
**Pull request by wenmin92**
[XnResize](https://www.xnview.com/en/xnresize/): A fast, powerful and free cross-platform image resizer.

### CODE REVIEW

File: bucket/xnresize.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+    "version": "1.00",
+    "description": "A fast, powerful and free cross-platform image resizer",
```
Comment (Ash258) [inline bucket/xnresize.json]:
```suggestion
    "description": "A fast and powerful image resizer",
```

File: bucket/xnresize.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    },
+    "extract_dir": "XnResize",
+    "bin": "XnResize.exe",
+    "shortcuts": [
+        [
+            "XnResize.exe",
+            "XnResize"
+        ]
+    ],
+    "checkver": "XnResize ([\\d.]+)",
```
Comment (Ash258) [inline bucket/xnresize.json]:
```suggestion
    "checkver": ">XnResize ([\\d.]+)\\s+:",
```

**Comment by wenmin92** [comment]
> Use Copy over persist method for `xnresize.ini`

Sorry, I don't understand what it means.

**Comment by Ash258** [comment]
1. Copy the file form persist on installation
1. Copy the file to persist on uninstallation

**Comment by wenmin92** [comment]
The `xnresize.ini` file is locate in `$APPDATA\XnResize`, copy this file to persist folder? why I need to do this?

**Comment by Ash258** [comment]
1. Open xnresize
1. Change configuration
1. Close xnresize
1. Open folder APPDATA\Xnresize
1. Copy xnresize.ini to $dir
1. Rename folder APPDATA\Xnresize to xnresize.backup
1. Launch xnresize
1. Change settings
1. Notice that configuration file was changed in $dir

**Comment by wenmin92** [comment]
OK, understand.

**Comment by wenmin92** [comment]
The `xnresize.ini` file seems not support hard link. XnResize will break the link.

**Comment by Ash258** [comment]
As every QT application. Thats why I tell you to use copy over method

File: bucket/xnresize.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        "64bit": {
+            "url": "https://download.xnview.com/XnResize-win-x64.zip",
+            "hash": "472bd46899743f7f6cb602fc09c92185ca3b287a9841a152e545174dc0cd78a7"
+        }
+    },
+    "pre_install": [
+        "if ((Test-Path \"$env:APPDATA\\XnResize\\xnresize.ini\")) { Move-Item -Force -Path \"$env:APPDATA\\XnResize\" -Destination \"$dir\"}",
+        "elseif (Test-Path \"$persist_dir\\xnresize.ini\") { Copy-Item \"$persist_dir\\xnresize.ini\" \"$dir\" -Force }",
+        "else { New-Item -Path \"$dir\\xnresize.ini\" -ItemType 'file' | Out-Null }"
+    ],
```
Comment (Ash258) [inline bucket/xnresize.json]: You do not need appdata. If the file is created next to executable it is used....

Move pre_install after extract_dir..

```suggestion
    "pre_install": [
        "if (-not (Test-Path \"$persist_dir\\xnresize.ini\")) {",
        "    New-Item \"$persist_dir\" -ItemType Directory -ErrorAction SilentlyContinue -Force | Out-Null",
        "    Set-Content \"$dir\\xnresize.ini\" '<FILL SOME BATCH, which i cannot remember>', 'update=0' -Encoding Ascii",
        "} else {",
        "    Copy-Item \"$persist_dir\\xnresize.ini\" \"$dir\"",
        "}"
    ],
```

File: bucket/xnresize.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        [
+            "XnResize.exe",
+            "XnResize"
+        ]
+    ],
+    "uninstaller": {
+        "script": [
+            "New-Item \"$persist_dir\" -ItemType Directory -ErrorAction SilentlyContinue | Out-Null",
+            "if (Test-Path \"$dir\\xnresize.ini\") { Copy-Item \"$dir\\xnresize.ini\" \"$persist_dir\" -Force }"
+        ]
```
Comment (Ash258) [inline bucket/xnresize.json]:
```suggestion
        "script": "Copy-Item \"$dir\\xnresize.ini\" \"$persist_dir\" -ErrorAction SilentlyContinue -Force",
```

File: bucket/xnresize.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        "else { New-Item -Path \"$dir\\xnresize.ini\" -ItemType 'file' | Out-Null }"
+    ],
+    "extract_dir": "XnResize",
+    "bin": "XnResize.exe",
+    "shortcuts": [
+        [
+            "XnResize.exe",
+            "XnResize"
+        ]
+    ],
```
Comment (Ash258) [inline bucket/xnresize.json]: After uninstaller

File: bucket/xnresize.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        }
+    },
+    "extract_dir": "XnResize",
+    "pre_install": [
+        "if (-not (Test-Path \"$persist_dir\\xnresize.ini\")) {",
+        "    New-Item \"$persist_dir\" -ItemType Directory -ErrorAction SilentlyContinue -Force | Out-Null",
+        "    Set-Content \"$dir\\xnresize.ini\" '[BatchResize]', 'update=0' -Encoding Ascii",
+        "} else {",
+        "    Copy-Item \"$persist_dir\\xnresize.ini\" \"$dir\"",
+        "}"
```
Comment (Ash258) [inline bucket/xnresize.json]: Only visualize tweak and saving of one negate

```suggestion
        "if (Test-Path \"$persist_dir\\xnresize.ini\") {",
        "    Copy-Item \"$persist_dir\\xnresize.ini\" \"$dir\"",
        "} else {",
        "    New-Item \"$persist_dir\" -ItemType Directory -ErrorAction SilentlyContinue -Force | Out-Null",
        "    Set-Content \"$dir\\xnresize.ini\" '[BatchResize]', 'update=0' -Encoding Ascii",
        "}"
```

**Pull request merged**